tests/unit/hykkt/HykktPermutationTests.hpp

```diff
@@ -1,130 +1,280 @@
-        pc.addHInfo(a_i, a_j);
-        pc.addJInfo(a_i, a_j, n, m);
-        pc.addJtInfo(a_i, a_j);
+        // n_hes = m_jac = 3, n_jac = 2
+        int n       = 3;
+        int m       = 2;
+        int nnz_hes = 6;
+        int nnz_jac = 4;
+
+        int  h_perm[3] = {2, 0, 1};
+        int* perm;
+        if (memspace_ == memory::HOST)
+        {
+          perm = h_perm; // already on host
+        }
+        else
+        {
+          mem_.allocateArrayOnDevice(&perm, n);
+          mem_.copyArrayHostToDevice(perm, h_perm, n);
+        }
+
+        matrix::Csr* hes    = new matrix::Csr(n, n, nnz_hes);
+        matrix::Csr* jac    = new matrix::Csr(m, n, nnz_jac);
+        matrix::Csr* jac_tr = new matrix::Csr(n, m, nnz_jac);
+        getTestData(hes, jac, jac_tr);
+
+        // correct results
+        int hes_prc_i[4] = {0, 2, 4, 6};
+        int hes_prc_j[6] = {1, 2, 0, 1, 0, 2};
+
+        int jac_pc_i[3] = {0, 2, 4};
+        int jac_pc_j[4] = {0, 1, 0, 2};
+
+        int jac_tr_pr_i[4] = {0, 2, 3, 4};
+        int jac_tr_pr_j[4] = {0, 1, 0, 1};
+
+        int result_prc_i[4] = {};
+        int result_prc_j[6] = {};
+        int result_pr_i[4]  = {};
+        int result_pr_j[4]  = {};
+        int result_pc_j[4]  = {};
+
+        ReSolve::hykkt::Permutation pc = ReSolve::hykkt::Permutation(n, m, nnz_hes, nnz_jac, memspace_);
+
+        pc.addHInfo(hes);
+        pc.addJInfo(jac);
+        pc.addJtInfo(jac_tr);
+
         pc.addPerm(perm);
         pc.invertPerm();
 
         // Test RC permutation
-        pc.vecMapRC(b_i, b_j);
-        printf("Comparing RC permutation\n");
-        for (int i = 0; i < n + 1; i++) // Loop over row pointers (n+1)
+        pc.vecMapRC(result_prc_i, result_prc_j);
+        printf("Comparing RC permutation of H\n");
+        bool flagrc = verifyResults(hes_prc_i, result_prc_i, n + 1);
+        flagrc &= verifyResults(hes_prc_j, result_prc_j, nnz_hes);
+        printf(!flagrc ? "RC permutation failed\n" : "RC permutation passed\n");
+
+        // Test R permutation
+        pc.vecMapR(result_pr_i, result_pr_j);
+        printf("Comparing R permutation of J_tr\n");
+        bool flagr = verifyResults(jac_tr_pr_i, result_pr_i, m + 1);
+        flagr &= verifyResults(jac_tr_pr_j, result_pr_j, nnz_jac);
+        printf(!flagr ? "R permutation failed\n" : "R permutation passed\n");
+
+        // Test C permutation
+        pc.vecMapC(result_pc_j);
+        printf("Comparing C permutation of J\n");
+        bool flagc = verifyResults(jac_pc_j, result_pc_j, nnz_jac);
+        printf(!flagc ? "C permutation failed\n" : "C permutation passed\n");
+
+        double hes_prc_v[6]   = {4, 5, 1, 0, 3, 2};
+        double jac_pc_v[4]    = {1, 0, 3, 2};
+        double jac_tr_pr_v[4] = {2, 3, 0, 1};
+
+        // Test mapIndex on H
+        double* result_prc_v = allocateArray(nnz_hes);
+        pc.mapIndex(ReSolve::hykkt::PERM_HES_V, hes->getValues(memspace_), result_prc_v);
```

Comment:
```suggestion
        pc.mapIndex(ReSolve::hykkt::PERM_HES_V, hes.getValues(memspace_), result_prc_v);
```
for moving the matrix off the heap